File: AspNetMvcUnleashed/0672329980_source/Chapter12/CS/MvcApplication1/MvcApplication1/Controllers/JillController.cs
using System.Security.Principal;
using System.Web.Mvc;

namespace MvcApplication1.Controllers
{
    public class JillController : Controller
    {

        public ActionResult Index(IPrincipal user)
        {
            if (user.Identity.Name != "Jill")
                return new HttpUnauthorizedResult();

            return View();
        }

    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter12/VB/MvcApplication1/MvcApplication1/Controllers/JillController.vb
Imports System.Security.Principal

Public Class JillController
    Inherits Controller

    Public Function Index(ByVal user As IPrincipal) As ActionResult
        If user.Identity.Name <> "Jill" Then
            Return New HttpUnauthorizedResult()
        End If

        Return View()
    End Function

End Class
